File: app/src/main/java/com/filemanager/fileexplorer/myfile/cleanmaster/free/Iinterface/BackgroundTask.java
package com.filemanager.fileexplorer.myfile.cleanmaster.free.Iinterface;

import android.content.Context;
import android.os.Handler;
import android.os.Looper;
import android.os.Message;

import com.filemanager.fileexplorer.myfile.cleanmaster.free.Activity.customview.Dialog_thread;


public class BackgroundTask {
    private Context activity;

    public BackgroundTask(Context context) {
        this.activity = context;
    }

    public void Handleloop(final HandleLooper handleLooper) {
        final Dialog_thread dialog_thread = new Dialog_thread(this.activity);
        dialog_thread.show();
        new Handler(Looper.getMainLooper()) {
            @Override
            public void handleMessage(Message message) {
                handleLooper.update();
                dialog_thread.dissmiss();
            }
        }.obtainMessage(111, "parameter").sendToTarget();
    }
}

File: app/src/main/java/com/demo/filemanager/Iinterface/BackgroundTask.kt
package com.demo.filemanager.Iinterface

import android.content.Context
import android.os.Handler
import android.os.Looper
import android.os.Message
import com.demo.filemanager.Activity.customview.Dialog_thread

class BackgroundTask(private val activity: Context) {
    fun Handleloop(handleLooper: HandleLooper) {
        val dialog_thread = Dialog_thread(activity)
        dialog_thread.show()
        object : Handler(Looper.getMainLooper()) {
            override fun handleMessage(message: Message) {
                handleLooper.update()
                dialog_thread.dissmiss()
            }
        }.obtainMessage(111, "parameter").sendToTarget()
    }
}
